framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1526,9 +1520,7 @@ int pcm_mmap_transfer(struct pcm *pcm, const void *buffer, unsigned int bytes)
 		if ((unsigned int)avail == 0) {
 			err = pcm_wait(pcm, -1);
 			if (err < 0) {
-				pcm->prepared = 0;
-				pcm->running = 0;
-				return oops(pcm, errno, "wait error: avail 0x%x\n", avail);
+				return err;
```

Comment:
current code is right.
Also we jsut return xrun msg without change value of pcm->prepared and pcm->running in pcm_writei, pcm_readi.
Could you check this once again?

Reply:
I will change this after we decide on how to handle the xrun. I think the pcm will continue to run even after an xrun.